src/Microsoft.ML.Data/Scorers/FeatureContributionCalculation.cs

```diff
@@ -0,0 +1,449 @@
+
+            var featureGetter = input.GetGetter<TSrc>(colSrc);
+            var map = Predictor.GetFeatureContributionMapper<TSrc, VBuffer<float>>(_topContributionsCount, _bottomContributionsCount, _normalize);
+
+            var features = default(TSrc);
+            var contributions = default(VBuffer<float>);
+            return
+                (ref ReadOnlyMemory<char> dst) =>
+                {
+                    featureGetter(ref features);
+                    map(in features, ref contributions);
+                    var editor = VBufferEditor.CreateFromBuffer(ref contributions);
+                    var indices = contributions.IsDense ? Utils.GetIdentityPermutation(contributions.Length) : editor.Indices;
+                    var values = editor.Values;
+                    var count = values.Length;
+                    var sb = new StringBuilder();
+                    GenericSpanSortHelper<int>.Sort(indices, values, 0, count);
+                    for (var i = 0; i < count; i++)
+                    {
+                        var val = values[i];
+                        var ind = indices[i];
+                        var name = GetSlotName(ind, slotNames);
+                        sb.AppendFormat("{0}: {1}, ", name, val);
+                    }
+
+                    if (sb.Length > 0)
+                    {
+                        _env.Assert(sb.Length >= 2);
+                        sb.Remove(sb.Length - 2, 2);
+                    }
+
+                    dst = new ReadOnlyMemory<char>(sb.ToString().ToCharArray());
+                };
+        }
+
+        private ValueGetter<VBuffer<float>> GetValueGetter<TSrc>(Row input, int colSrc)
+        {
+            Contracts.AssertValue(input);
+            Contracts.AssertValue(Predictor);
+
+            var featureGetter = input.GetGetter<TSrc>(colSrc);
+
+            // REVIEW: Scorer can do call to Sparicification\Norm routine.
+
+            var map = Predictor.GetFeatureContributionMapper<TSrc, VBuffer<float>>(_topContributionsCount, _bottomContributionsCount, _normalize);
+            var features = default(TSrc);
+            return
+                (ref VBuffer<float> dst) =>
+                {
+                    featureGetter(ref features);
+                    map(in features, ref dst);
+                };
+        }
+
+        private static void CheckSchemaValid(IExceptionContext ectx, RoleMappedSchema schema,
+            IFeatureContributionMapper predictor)
+        {
+            Contracts.AssertValue(ectx);
+            ectx.AssertValue(schema);
+            ectx.AssertValue(predictor);
+
+            // REVIEW: Check that Features column is present and is of correct size and item type.
+        }
+    }
+
+    /// <summary>
+    /// Maps a schema from input columns to output columns. Keeps track of the input columns that are needed for the mapping.
+    /// </summary>
+    internal sealed class RowMapper : ISchemaBoundRowMapper
```

Comment:
>RowMapper [](start = 26, length = 9)

Again a top level class with a very, very high level name `RowMapper`, inside one of our most central assemblies. Not sure what's going on.